commands/christmaspatrol.js

```diff
@@ -0,0 +1,22 @@
+module.exports = function (bb8,options) {
+ var radius = options.radius; //mm
+ var distance = 2*Math.PI*radius;
+ var colors = ["red","green"];
+ var i = 0;
+ return setInterval(function() {
+    var direction = 30;
+    bb8.roll(Math.floor(distance/12),direction*i);
+    var chosen = (function(){
+      if (Math.random() > 0.5) {
+        return 1;
+       } else {
+        return 0;
+       }
```

Comment:
I'm not sure if you prefer this, but i typically write functions with conditional like this 

```
function() {
    
    if (Math.random() > 0.5) {
        return 1;
    }

    return 0;
}
```

Reply:
couldn't this become: 
```
function() {
  return Math.random() > 0.5 ? 1 : 0
}

// or using ES6 arrow syntax:
() => Math.random() > 0.5 ? 1 : 0
```

Reply:
You could also change this logic entirely to just pull out a random element from the `colors` var. That way, if you wanted to add new colours in the future you'd be able to do it by just adding a new colour into the `colors` array.

You could do it like this:

```js
function getRandomElement(arr) {
   return arr[Math.floor(Math.random() * arr.length)];
}
```

Or just use `lodash` and cheat:

```js
// no need for a function as that's exactly what this does:
_.sample(colors)
```

Reference: https://lodash.com/docs/4.17.5#sample

Personally I'd probably just use lodash.

Reply:
Lodash is already part of the project as well so i've got no issue in using it. Nice suggestion though i'm not sure yellow is a christmas colour 😄